modules/andes-core/broker/src/main/java/org/wso2/andes/store/file/FileMessageStoreImpl.java

```diff
@@ -0,0 +1,1364 @@
+                bytes(Long.toString(metadata.getExpirationTime())));
+    }
+
+    /**
+     * add message content details to the transaction
+     *
+     * @param messagePart message content chunk which needed to be stored
+     * @param transaction transaction which accumulates operations to commit
+     *                    <p>
+     *                    Following keys will be created
+     *                    <p>
+     *                    MESSAGE.$message_id.$offset.MESSAGE_CONTENT
+     */
+    private void addContent(AndesMessagePart messagePart, Transaction transaction) {
+        String messageContentIdentifier = generateKey(MESSAGE_CONTENT, MESSAGE,
+                Long.toString(messagePart.getMessageID()), Long.toString(messagePart.getOffset()));
+
+        transaction.put(bytes(messageContentIdentifier), messagePart.getData());
+    }
+
+    /**
+     * create mapping between the destination and the message metadata
+     *
+     * @param metadata    message metadata which is needed to be mapped
+     * @param transaction transaction which accumulates operations to commit
+     *                    <p>
+     *                    Following keys will be created
+     *                    <p>
+     *                    DESTINATION.$destination_name.$message_id.MESSAGE_METADATA
+     *                    DESTINATION.$destination_name.MESSAGE_COUNT
+     */
+    private void createQueueMetaDataMapping(AndesMessageMetadata metadata, String queueName, Transaction transaction) {
+        String queueMessageMetaDataIdentifier = generateKey(MESSAGE_METADATA, DESTINATION, queueName,
+                Long.toString(metadata.getMessageID()));
+        String queueMessageCountIdentifier = generateKey(MESSAGE_COUNT, DESTINATION, queueName);
+
+        Long queueMessageCount;
+
+        if (asString(transaction.getKey(queueMessageCountIdentifier)) == null) {
+            queueMessageCount = Long.parseLong(asString(brokerStore.get(bytes(queueMessageCountIdentifier))));
+        } else {
+            queueMessageCount = Long.parseLong(asString(transaction.getKey(queueMessageCountIdentifier)));
+        }
+        queueMessageCount++;
+        transaction.setKey(queueMessageCountIdentifier, bytes(Long.toString(queueMessageCount)));
+
+        transaction.put(bytes(queueMessageMetaDataIdentifier), metadata.getMetadata());
+        transaction.put(bytes(queueMessageCountIdentifier), bytes(Long.toString(queueMessageCount)));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void storeMessagePart(List<AndesMessagePart> partList) throws AndesException {
+        Transaction tx = new Transaction(brokerStore);
+
+        try {
+            for (AndesMessagePart messagePart : partList) {
+                addContent(messagePart, tx);
+            }
+            tx.commit(brokerStore);
+        } catch (DBException e) {
+            log.error("Message parts storing failed", e);
```

Comment:
Shouldn't transaction rollback here?

Reply:
According to the transaction implementation, the changes will be applied to the database once the transaction is committed. Hence, if the transaction fails changes will not be applied and rollback is not required.